src/main/cpp/src/decimal_utils.hpp

```diff
@@ -16,45 +16,39 @@
 
 #include <cudf/column/column_view.hpp>
 #include <cudf/table/table.hpp>
+#include <cudf/utilities/default_stream.hpp>
+
+//
```

Comment:
Comment with no text?  Should either remove these or add a comment.  I'm assuming this is to delineate sections of the includes (IMO not necessary, pretty obvious from include paths or lack thereof), but without text it's not clear what the intent is.

Applies to everywhere this was added in this PR.

Reply:
This is due to the issue with clang-format that messes up the header order. I have to use an empty comment line to prevent header reordering.

Reply:
Does it need to be empty, or can the comment say "// this is here to avoid clang-format from rearranging the header files".  Also wondering why we're explicitly avoiding the recommended header order?

Reply:
The current clang-format config combines headers together and doesn't respect the conventional "near to far"  order. That means we should include our header first, then library headers, then builtin headers. By using "near to far" order, we can ensure that most headers have their own dependencies included themselves.

We probably should change the config a little bit to account for that.

Reply:
Ideally if we switch to use `.clang-format` style from cudf then we don't have to worry about this as cudf config can avoid this mixing header issue.

Reply:
There is another option: fix the style to preserve header grouping: https://github.com/NVIDIA/spark-rapids-jni/pull/1584. With that, the header organization is totally up to the code author.